app/models/san_address.rb

```diff
@@ -0,0 +1,14 @@
+class SanAddress < ApplicationRecord
+  include ProviderObjectMixin
+
```

Comment:
You need `include NewWithTypeStiMixin`

Reply:
@agrare, adding `NewWithTypeStiMixin` cause a wierd bug that I'm having difficulties to debug.
`[----] I, [2020-12-23T13:01:30.670638 #15268:2af34be3795c]  INFO -- : EMS: [autosd-219], id: [2] Saving EMS Inventory...
[----] E, [2020-12-23T13:01:30.841971 #15268:2af34be3795c] ERROR -- : Error when saving InventoryCollection:<SanAddress> with strategy: , saver_strategy: default, targeted: false. Message: undefined method `constantize' for #<Class:0x000055e6a6d8f698>
Did you mean?  constants`

![Screenshot 2020-12-23 135127](https://user-images.githubusercontent.com/53213107/102996191-4f395600-452b-11eb-9422-d46d1aff455c.jpg)

the link http://coderrr.wordpress.com/2008/04/22/building-the-right-class-with-sti-in-rails/ inside `new_with_type_sti_mixin.rb` is not working, can you explain me what the rational of using it, or maybe you are familiar with the bug?

Reply:
Without seeing the backtrace my best guess is that you're passing in the class, not the class name to the `:type` attribute of the inventory object that you're building.

E.g. make sure you're doing `persister.san_addresses.build(:type => "NvmeAddress")` not `.build(:type => NvmeAddress)`

Reply:
@agrare good guess, this was the issue, thanks.